### PR TITLE
Add "*spec.ExpandOptions" parameter to Expanded().

### PR DESCRIPTION
**Pull request by prettyxw**
closes: #7 
Signed-off-by: Peng Jingwen <pengsrc@icloud.com>

### CODE REVIEW

**Comment by prettyxw** [comment]
This pull request is relevant to https://github.com/go-openapi/spec/pull/15.

File: spec.go [hunk 1]
```diff
@@ -134,12 +134,12 @@ func Analyzed(data json.RawMessage, version string) (*Document, error) {
 }
 
 // Expanded expands the ref fields in the spec document and returns a new spec document
-func (d *Document) Expanded() (*Document, error) {
+func (d *Document) Expanded(options *spec.ExpandOptions) (*Document, error) {
```
Comment (casualjim) [inline spec.go]: Would it be possible to make this ExpandedWithOptions and keep the Expanded method without arguments.
Then the expand options could have as default option the dir of the path of the spec it loaded

Comment (prettyxw) [inline spec.go]: Got it. Fixed that.

**Comment by casualjim** [comment]
I really like these changes but when I use them the following  tests never finish:

https://github.com/go-openapi/validate/blob/master/spec_test.go#L36-L149

All of them are invalid specs though, when I skip those tests everything else passes, do you have any idea why that would be?

**Comment by prettyxw** [comment]
It's my fault to only test "loads" and "spec" repo, I'll check other repos as well.

**Comment by casualjim** [comment]
k I'll leave this open for a short while, I think it fixes as many bugs as the tests I've had to skip. Since all the tests that fail are validation cases, perhaps it's still a good trade-off, but will wait to see what you find. 

With this PR go-swagger is the first library on go that fully supports json schema and ref resolving so very happy with it :+1:

**Comment by prettyxw** [comment]
Hi, the latest commit on spec repo should pass all the test in validate repo.

**Pull request merged**